[PS-107] Add remove option for bearcard

kangwooklee29 opened the pull request:
Changelog
====
- Adds remove option for bearcard, to change the card status into "Deleted" using this command. 
  - This option uses the script in `bin/change_card_status.sh`, already used by the workflow, so the lines of change is very few.
- Adds the code to print the success/failure message for `bin/change_card_status.sh`, to show the result of the script clearly.
- Refactors `bin/change_card_status.sh` & `bearcard` to change the API endpoint when fetching all the cards. Previously it fetched all the list IDs first and then it found the card ID using the list IDs, since I didn't know the API endpoint that can fetch all the cards even though I don't know the specific list ID. Now it directly fetches all the cards not using the list IDs.
- Fixes a bug when running `bearcard --update-key` by introducing a function `check_config_file` in bearcard. Previously it didn't run properly since it needed the config file to run the code, due to my mistake.


Detailed Description
====
After this commit is merged, you can change the card status into "Deleted" by running `bearcard -r "Any string in the card name"` simply.

sae1013 commented:
How about using RegExr to find target ticket which should be removed ? 
i'm afraid of same card name or same substring 
```shell
^ps-[0-9]+$ 
```

sae1013 commented:
when i ran bear command with -r option, i found this error meesage. 
```shell
/bin  ➦ a067b23 ±  bearcard -r ps-98
/Users/minwoo/repository/bearalgorithm/bin/command/bearcard: line 156: realpath: command not found
usage: dirname string [...]
usage: dirname string [...]
Trying to delete "PS-98"...
/Users/minwoo/repository/bearalgorithm/bin/command/bearcard: line 162: /change_card_status.sh: No such file or directory
```
i think SCRIPT_PATH or PARENT_DIR is incorrect right? 
do i need additional setup?
thanks !!

sae1013 commented:
Appreciate your hard work

kangwooklee29 commented:
1. I changed it to use the regex you wrote. I think your concerns are reasonable.
2. It was a bug, so I fixed it. Actually I wrote & tested the commit in Windows Git Bash environment and it worked well but `realpath` doesn't work in macOS environment, because `realpath` is for GNU/Linux but macOS is from BSD. but actually we are already using the code that works in both environments, `"$(cd "$(dirname "$0")"; pwd -P)"`, in `bin/change_card_status.sh`. So I replaced `realpath` with it.

Thanks for your detailed review. You are a good reviewer and I think you're a main contributor to this shell script command!
